crawl-ref/source/dat/des/variable/mini_monsters.des

```diff
@@ -1229,6 +1229,76 @@ x<<xxxxxxxxxxxxxxxxxxcx
 x&&xxxxxxxxxxxxxxxxxxxx
 ENDMAP
 
+# I'd apologize for this pun, but I'm not actually sorry.
+NAME:    nicolae_half_adder
+TAGS:    transparent patrolling
+DEPTH:   D:2-6
+KMONS:   1 = adder
+KFEAT:   !O = c
+SHUFFLE: pqrs/O!!O/!O!O/!!O!
+SUBST:   pqrs = O
+: nsubst( "1 = " .. you.depth() .. "=1 / 1=1. / ." )
+TILE:    ! = DNGN_TRANSPARENT_STONE_WHITE
+TILE:    O = DNGN_TRANSPARENT_STONE_DARKGRAY
+COLOUR:  ! = white
+COLOUR:  O = magenta
+MAP
+.....................
+..p.........c.c....1.
+.pppppppppppcpcc..r..
+..p.p.1..1..1.%ccrrr.
+.1...p....qqcqcc..r..
+....1.p..q.1c.c..1...
+.......pq............
+.......qp............
+....1.q..p.1cccc.1...
+.1...q....ppccccc.s..
+..q.q.1..1..1.%ccsss.
+.qqqqqqqqqqqccccc.s..
+..q.........cccc...1.
+.....................
+ENDMAP
+
+# A similar idea didn't cut it as a Swamp rune vault, so let's resurrect it as
+# a lower-Dungeon vault. The gimmick, of course, is that each room has
+# 27 hydra heads, just arranged on a different number of hydra bodies.
+# Only places if this game doesn't have Swamp. This way, there's no incentive
+# to dive down through D in the hopes of finding this vault before the Swamp
+# end is generated to keep the Lernaean Hydra from being generated in the
+# rune vault.
+NAME:   nicolae_hydra_factorization
+TAGS:   no_monster_gen no_trap_gen
+DEPTH:  D:12-
+veto {{
+   return (crawl.game_started() and dgn.br_exists("Swamp"))
+}}
+# You can see the loot room from just outside the front door to the
+# Lernaean Hydra room (assuming you have at least 6 LOS range), so you can
+# decide if you want to go in. The loot is geared towards good stuff,
+# including a guaranteed randart, because otherwise who would go in?
+KITEM:  * = nothing / |, |, |, |, |, gold good_item, \
```

Comment:
I would use a different glyph since you're not actually placing `star_item` at all. Probably just do `KITEM:   d = ...` and use the first item slot.